### PR TITLE
Remove pipline aggs from agg result tree

### PR DESCRIPTION
**Pull request by nik9000**
This removes pipeline aggregators from the aggregation result tree
except for a single field used for backwards compatibility with pre-7.8
versions of Elasticsearch. That field isn't populated unless we are
serializing to pre-7.8 Elasticsearch. So, good news! We no longer build
pipeline aggregators on the data node. Most of the time.

### CODE REVIEW

**Comment by nik9000** [comment]
@polyfractal, this one is also all deletes. Very careful little deletes.

File: server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java [hunk 1]
```diff
@@ -223,13 +223,6 @@ public int countAggregators() {
         return factories.length;
     }
 
-    /**
-     * @return the number of pipeline aggregator factories
-     */
-    public int countPipelineAggregators() {
```
Comment (polyfractal) [inline server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java:229]: Haha was this even used anywhere?  I don't I've seen it before :man_facepalming:

**Review by polyfractal — APPROVED** [review]
LGTM, albeit with only a fast skim.  Eyes started to go fuzzy after a hundred files :)  But the deletes are all simple so and there's not really a better way to do this!  Thanks for tackling! :+1:

File: server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java (hunk 1 above)
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java:229]: Not any more!

**Comment by nik9000** [comment]
@elasticmachine update branch

**Comment by nik9000** [comment]
run elasticsearch-ci/packaging-sample-unix-docker

**Pull request merged**